mlflow/store/artifact/databricks_artifact_repo.py

```diff
@@ -444,188 +436,16 @@ def list_artifacts(self, path=None):
         return infos
 
     def _download_file(self, remote_file_path, local_path):
-        """
-        _download_file is unused in this repository's implementation and is only defined out
-        of necessity because it is an abstract method in the `ArtifactRepository` base class
-        """
-
-    def download_artifacts(self, artifact_path, dst_path=None):
-        """
-        Parallelized implementation of `download_artifacts` for Databricks.
-        """
-
-        # Represents an in-progress file artifact download to a local filesystem location
-        InflightDownload = namedtuple(
-            "InflightDownload",
-            [
-                # The artifact path, given relative to the repository's artifact root location
-                "src_artifact_path",
-                # The local filesystem destination path to which artifacts are being downloaded
-                "dst_local_path",
-                # A future representing the artifact download operation
-                "download_future",
-            ],
+        run_relative_remote_file_path = posixpath.join(
+            self.run_relative_artifact_repo_root_path, remote_file_path
+        )
+        read_credentials = self._get_read_credential_infos(
+            run_id=self.run_id, paths=[run_relative_remote_file_path]
+        )
+        # TODO(yogita): Assert for len(read_credentials) == 1
```

Comment:
Did we want to do this TODO in this PR? Also, why does this have to be of length 1? (not super familiar with `get_read_credential_infos`)

Reply:
Added comment in the code to explain this.

The _get_read_credential_infos uses _get_credential_infos which supports getting credentials for batch of artifacts. Here we are reading credential for a single artifact, so expect only one entry in the response. @dbczumar please review and correct in case I got this wrong.

Reply:
That's right!